testsuite/tests/typing-jkind-bounds/subsumption/modalities.ml

```diff
@@ -0,0 +1,320 @@
+         value mod portable
+         because of the definition of t at line 4, characters 2-38.
+|}]
+
+module M : sig
+  type ('a, 'b) t : value mod portable with 'b
+end = struct
+  type ('a, 'b) t : value mod portable with 'a @@ portable with 'b @@ contended
+end
+[%%expect {|
+module M : sig type ('a, 'b) t : value mod portable with 'b end
+|}]
+
+module M : sig
+  type 'a t : immutable_data with 'a @@ portable
+end = struct
+  type 'a t : immutable_data with 'a @@ portable with 'a
+end
+[%%expect {|
+Lines 3-5, characters 6-3:
+3 | ......struct
+4 |   type 'a t : immutable_data with 'a @@ portable with 'a
+5 | end
+Error: Signature mismatch:
+       Modules do not match:
+         sig type 'a t : immutable_data with 'a end
+       is not included in
+         sig type 'a t : immutable_data with 'a end
+       Type declarations do not match:
+         type 'a t : immutable_data with 'a
+       is not included in
+         type 'a t : immutable_data with 'a
+       The kind of the first is immutable_data with 'a
+         because of the definition of t at line 4, characters 2-56.
+       But the kind of the first must be a subkind of immutable_data with 'a
+         because of the definition of t at line 2, characters 2-48.
+|}]
+
+module M : sig
+  type 'a t : immutable_data with 'a @@ portable
+end = struct
+  type 'a t : immutable_data with 'a with 'a @@ portable
+end
+[%%expect {|
+Lines 3-5, characters 6-3:
+3 | ......struct
+4 |   type 'a t : immutable_data with 'a with 'a @@ portable
+5 | end
+Error: Signature mismatch:
+       Modules do not match:
+         sig type 'a t : immutable_data with 'a end
+       is not included in
+         sig type 'a t : immutable_data with 'a end
+       Type declarations do not match:
+         type 'a t : immutable_data with 'a
+       is not included in
+         type 'a t : immutable_data with 'a
+       The kind of the first is immutable_data with 'a
+         because of the definition of t at line 4, characters 2-56.
+       But the kind of the first must be a subkind of immutable_data with 'a
+         because of the definition of t at line 2, characters 2-48.
+|}]
+
+module M : sig
+  type 'a t : immutable_data with 'a
+end = struct
+  type 'a t : immutable_data with 'a @@ portable with 'a
+end
+[%%expect {|
+module M : sig type 'a t : immutable_data with 'a end
+|}]
+
+type 'a u : value mod uncontended with 'a @@ global
+type 'a t : value mod global = 'a u
+[%%expect {|
+type 'a u : value mod uncontended with 'a
+Line 2, characters 0-35:
+2 | type 'a t : value mod global = 'a u
+    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Error: The kind of type "'a u" is value mod uncontended with 'a
+         because of the definition of u at line 1, characters 0-51.
+       But the kind of type "'a u" must be a subkind of value mod global
+         because of the definition of t at line 2, characters 0-35.
+|}]
+
+module M : sig
+  type 'a t : immutable_data with 'a @@ contended portable
```

Comment:
Can we try this the other way around? That way should work.

Reply:
done